3rdparty/CMakeLists.txt

```diff
@@ -0,0 +1,3 @@
+add_subdirectory(apg)
+add_subdirectory(jpeg-compressor)
+add_subdirectory(lodepng)
```

Comment:
Does it make sense to use external libraries instead of the vendored sources?
A user of basic_universal might itself use lodepng.
This would fail linking because of duplicate symbols.
The same applies to apg and jpeg-compressor.

There is no problem if these are built in a completely private namespace.